pkg/mimir/cortex_tool.go

```diff
@@ -0,0 +1,39 @@
+package mimir
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+
+	"github.com/grafana/grizzly/pkg/config"
+)
+
+type CortexTool interface {
+	ExecuteCortexTool(args ...string) ([]byte, error)
```

Comment:
I think it should be named in a way where we could switch over to mimirtool by default. 

Maybe it should be abstracted more so that it's a "prometheus client" and instead of `ExecuteCortexTool` we have `GetRules` that gives us a `map[string][]PrometheusRuleGroup{}` and a `LoadRules` function, etc, and so then we can use whatever tool we want in the backend

Reply:
I did it as a wrapper to be able to "mock" `exec.LookPath` behaviour and use it in tests easily.

But its a good idea because I'm working in a small client that its doing exactly that you mention 😂.  Mimir exposes an API and it could be a good idea to be able to use any of the options.